IBM/carbon-addons-iot-react
# feat(TableCard): dynamic links

Closes #1125 

This PR relies on #1099 for the dynamic variables functionality

**Summary**

- Adds dynamic link functionality to TableCard's by using a new `linkTemplate` prop in the TableCard JSON
- Its used with:
```
linkTemplate: {
  href: 'string',   // the url
  target: 'string'    // optional to open in a new window or not. This will use the literal '_blank'
}
```

**Change List (commits, features, bugs, etc)**

- If the `linkTemplate` prop is present in a column, render it's cells with a Link with the `href` and `target` values
- Added values to existing test cases
- Added `with links` story
- Converted some styled components to scss

@tay1orjones @davidicus There is a new prop added with this PR

**Acceptance Test (how to verify the PR)**

- Go to https://deploy-preview-1126--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-tablecard--with-links and see the dynamic links in action
- Go to  https://deploy-preview-1126--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-tablecard--with-dynamic-variables see the dynamic links in action
- Additionally, try out the knobs and see how the links can be dynamically changed
